### PR TITLE
Anchors in simulation

### LINKED ISSUE #560: Modify the Simulation Engine to Simulate Anchored Activities
This is a follow-up to #449. That ticket is about modifying the database to store/validate Anchors. This ticket is about modifying Simulation to process Anchors so that it can start simulating them at the correct time.

### PR DESCRIPTION
**Pull request by Mythicaeda**
* **Tickets addressed:** Closes #560
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

This PR:
* Renames files associated with Activity Directives to align with the current terminology we use within Aerie (ie `ActivityInstance` -> `ActivityDirective`).
* Retrieves `anchorId` and `anchoredToStart` from the database and adds them to `ActivityDirective`
* Adds `StartOffsetReducer`, which takes the set of activities to be simulated and maps each one to the ActivityDirective that must finish simulating before we know its absolute start time.
* Modifies how `SimulationDriver` schedules activities so that, when an activity finishes, all activities depending on it to know their start time are scheduled.

**Breaking Changes:** 
* `SimulationDriver.simulate()` has a new parameter: `planDuration`.
* `SimulationDriver.simulate()`'s `schedule` parameter had its type changed from `Map<ActivityInstanceId, Pair<Duration, SerializedActivity>>` to `Map<ActivityDirectiveId, ActivityDirective>`.

## Verification

Tests were created to test `StartOffsetReducer` and how anchors are processed in simulation.
Tests were modified to account for `SimulationDriver.simulate`'s new parameter.

### CODE REVIEW

File: merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java [hunk 1]
```diff
@@ -136,11 +138,83 @@ void simulateTask(final MissionModel<Model> missionModel, final TaskFactory<Retu
...
+          task,
+          schedule,
+          resolved,
+          missionModel,
+          activityTopic
+      ));
+    }
+  }
+
+  private static <Model, Output> TaskFactory<Unit> makeTaskFactory(
```
Comment (Mythicaeda) [inline merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java:177]: Add comments

File: merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java [hunk 2]
```diff
@@ -136,11 +138,83 @@ void simulateTask(final MissionModel<Model> missionModel, final TaskFactory<Retu
...
+                  schedule,
+                  resolved,
+                  missionModel,
+                  activityTopic
+              ));
+              return TaskStatus.completed(Unit.UNIT);
+            }));
+      }
+      return TaskStatus.completed(Unit.UNIT);
+    });
```
Comment (adrienmaillard) [inline merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java]: I can't make sense of that section. Could you make it clearer by trading conciseness for readability ?

**Review by adrienmaillard — APPROVED** [review]
Nice work. 

I appreciate the effort with testing, it really helps build confidence for this kind of deep change.

File: merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java (hunk 2 above)
Comment (Mythicaeda) [reply merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java]: I've added comments and renamed some of the variables, including `$` and `$$`. Please let me know if there are sections that need further clarification!

**Pull request merged**